test/test_tokenizer.rb

```diff
@@ -84,6 +84,25 @@ def test_comments
     assert_equal %w(foo COMMENT""" bar), tokenize("foo\n\"\"\"\nComment\n\"\"\"\nbar")
     assert_equal %w(foo COMMENT''' bar), tokenize("foo\n'''\nComment\n'''\nbar")
 
+    assert_equal %w(foo COMMENT/-), tokenize("foo /- Comment -/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-Comment-/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /- \nComment\n -/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-\nComment\n-/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-- Comment -/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /--Comment-/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /--\nComment\n-/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-! Comment -/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-!Comment-/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-!\nComment\n-/")
+    assert_equal %w(foo COMMENT/- bar), tokenize("foo /-\nComment\n-/ bar")
+    assert_equal %w(foo COMMENT/- bar), tokenize("foo /-\nComment\n-/\nbar")
+    assert_equal %w(foo COMMENT/- bar), tokenize("foo\n/-\nComment\n-/\nbar")
+    assert_equal %w(foo COMMENT/- bar), tokenize("foo /- Comment /- Still Comment /- And Still Comment -/ bar")
```

Comment:
Lean does not parse comments like C, as `body /* comment /* comment */ body */ body`. It parser them like parentheses, building up a stack, as `body /- comment /- comment2 -/ comment -/ body`.

Reply:
Having said that, I think it's totally fine for this lexer to not actually support this style of comment, since it's only used for the classifier anyway; but the tests should make it clear in a comment that this is a deliberate compromise!

Reply:
In that case
```suggestion
    # nested comments are not processed correctly, but we don't care
    assert_equal %w(foo COMMENT/- comment3 -/ bar), tokenize("foo /- comment1 /- comment2 -/ comment 3 -/ bar")
```